orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -30,6 +30,12 @@ def __init__(self, state, output_info):
         self.state = state
         self.output_info = output_info
 
+def _validate_module_type(module):
```

Comment:
How would a free standing function support the following case?
* user A adds ONNX support and want a specific validation 
* user B adds torch190 with specific validation 
* user C adds tf.keras.Model support and also want a specific implementation

As a free standing function, ORTModule can only support one validation function at a time, regardless of how many TorchModule implementations we have. On the other hand, if _validate_module is part of GraphExecutionManager interface, each user can have its own validation and the factory will decide which one to use in run time.

Reply:
User would need the following implementations to (experimentally) support different types as of now:
- ```GraphExecutionManager```s for their respective types (i.e. there will be a different ```GraphExecutionManager``` for torch modules and a different one for ONNX modules. Users can implement their on validation on their version of ```GraphExecutionManager```. As far as different versions of torch module implementations go, we can decide to move this validation as required when we have made up our mind on the implementation/design.
- ```TorchModuleInterface```s for the respective types of models (torch, ONNX, tf...). This defines all the Torch module APIs like ```named_buffers```, ```state_dict``` and so on.
- And for now, users would need to override the ```TorchModuleFactory``` to monkeypatch its implementation to return a ```TorchModuleInterface``` for other model types. For example:
```py
import onnxruntime.training.ortmodule.ortmodule as ortmodule

class CustomTorchModuleFactory:
    def __call__(self, module):
        return TorchModuleForONNX(module)

# patch the factory
ortmodule.TorchModuleFactory = CustomTorchModuleFactory

model = ortmodule.ORTModule(onnx_model)
```

Let me know if this was not clear and I'll try to provide more details.

Reply:
My point is that although implementation may change in the future, our design must be as complete as possible in the present, to the best of our knowledge of possible/valid  scenarios.

The proposed solution artificially doesn't handle more than one custom torch module validation. There is no reason for such constrain, which can limit 1p' s experiments. 

Simply moving validate to GraphExecutionManager will allow as many torch module implementation and validation as the users want without conflicts, code inspection or asking for support.

Reply:
Proceeding with adding a new class ```GraphExecutionInterface``` which will serve as a base class for all the graph execution managers for torch and onnx. Moved ```_validate_module_type``` invocation to the constructor of the base class.